presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -164,6 +164,8 @@ public Page getOutput()
         // project results into output blocks
         List<Type> types = aggregates.stream().map(Aggregator::getType).collect(toImmutableList());
 
+        // output page will only be constructed once,
+        // so a new PageBuilder is constructed (instead of using PageBuilder.reset)
```

Comment:
In this case (and all the similar cases below) do we know the number of rows?  If we do, please set it in the constructor.  If not, update the comment to say, we do not know the number of rows.